discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -242,24 +243,24 @@ def get_purchase_metadata_from_memo(
             content_metadata = json.loads(memo)
             if "type" in content_metadata and "id" in content_metadata:
                 premium_conditions = None
-                if content_metadata["type"] == "TRACK":
-                    premium_conditions = (
-                        session.query(Track.premium_conditions)
+                if PurchaseType[content_metadata["type"].lower()] == PurchaseType.track:
+                    results = (
+                        session.query(Track.premium_conditions, Track.owner_id)
                         .filter(
                             Track.track_id == content_metadata["id"],
                             Track.is_current == True,
-                            # Track.premium_conditions["usdc_purchase"]["slot"] <= slot,
+                            Track.premium_conditions["usdc_purchase"] != None,
```

Comment:
Do we need an index on this?

Reply:
yeah it will, but i'm hesitant - I think ultimately I want to move the query to a different table with price histories and use _that_ and join on tracks after